container/webui/conf/openqa.ini

```diff
@@ -1,5 +1,5 @@
 [auth]
-method = OpenID
+method = Fake
```

Comment:
Well, the reason to use Fake is that if we use OpenID we will had to add more complexity. The OpenID gets as validation url the webui (http://webui instead of http://localhost:9526) And of course the token is not valid for the good url
```
naive_verify_failed_return: Direct contact invalidated ID provider response.
```
It could be solve creating a entry in /etc/hosts but forces the user to do this. Additionally the api_key should be created as in the bootstrap script. But I am not sure if will be valid before login.

On the contrary with Fake, the current docker-compose specification works.

In my opinion, the complexity should be moved to production deployments.

Reply:
As long as the instance is only accessible from the loopback device using fake auth should be fine. However, the problem of the wrong URL could likely be fixed by setting the `base_url` within the `[global]` section to the desired URL.

Reply:
> As long as the instance is only accessible from the loopback device using fake auth should be fine. However, the problem of the wrong URL could likely be fixed by setting the `base_url` within the `[global]` section to the desired URL.

I'd say try out if this works, assuming it takes a few minutes or so. If not, record the error - or if you tried it before, mention it here again.

I think we may have discussed this part but can't find a record of it. I'd consider proper auth a nice to have that can be a follow-up. Same category as TLS imo. I'd really like it, but it's not necessary for the quick setup story.

Reply:
I tried with OpenID as @Martchus said and works. And also, the default api_key exists from the begining. So keeping with OpenID.

Reply:
But I realized that I don't have the administration options 
![image](https://user-images.githubusercontent.com/11134265/109670268-3b395280-7b73-11eb-837e-e46ceb6b3e1b.png)

Reply:
Right, annoyingly the user (I literally mean a row in our users table) which is created on OpenID login won't have operator/admin permissions set.

You *could* run `ALTER TABLE users ALTER COLUMN is_operator SET DEFAULT 1;` and `ALTER TABLE users ALTER COLUMN is_admin SET DEFAULT 1;` to make every new user operator/admin. However, that would obviously mean everybody who can login can do everything which is likely also not what we want.

Not sure yet how to solve this in an elegant way.

Reply:
The very first user created is always admin, regardless of auth method, no?

Reply:
What do you (others) think? 
I propose to keep OpenID and comment in the commentary that is possible to change the auth to Fake to have full functionality. I don't know if is a good idea add more documentation with the SQL query. What do you think?

Reply:
False assertion. With OpenID we don't have any apikey until login, so worker cannot initizalizate. The reason of before was working was because I was keeping the DB from when I used Fake
So To not do this more complex, I am going to keep it with Fake

Reply:
You need to call the `/login` route once, yes, to create the first admin user

Reply:
> The very first user created is always admin, regardless of auth method, no?

Judging by the code that's not the case. For fake auth the permissions are hard-coded (so "Demo" is always admin) and for OpenID new users are never admin/operator.

And @ilausuch is also right about the API key/secret. When relying on OpenID we needed to generate the key for that user somehow *after* the first OpenID login.